packages/mobile/src/components/lineup-tile/CollectionTile.tsx

```diff
@@ -0,0 +1,229 @@
+import { isEqual } from 'lodash'
+
+import { useDispatchWeb } from 'app/hooks/useDispatchWeb'
+import { useNavigation } from 'app/hooks/useNavigation'
+import { useSelectorWeb } from 'app/hooks/useSelectorWeb'
+
+import { CollectionTileTrackList } from './CollectionTileTrackList'
+import { LineupTile } from './LineupTile'
+import { LineupItemProps } from './types'
+
+export const CollectionTile = (props: LineupItemProps) => {
+  const { uid } = props
+
+  const collection: Collection = useSelectorWeb(
+    state => getCollection(state, { uid }),
+    isEqual
+  )
+
+  const tracks: EnhancedCollectionTrack[] = useSelectorWeb(
+    state => getTracksFromCollection(state, { uid }),
+    isEqual
+  )
+
+  const user: User = useSelectorWeb(
+    state => getUserFromCollection(state, { uid }),
+    isEqual
+  )
+
+  if (!collection || !tracks || !user) {
+    console.warn(
+      'Collection, tracks, or user missing for CollectionTile, preventing render'
+    )
+    return null
+  }
+
+  if (collection.is_delete || user?.is_deactivated) {
+    return null
+  }
+
+  return (
+    <CollectionTileComponent
+      {...props}
+      collection={collection}
+      tracks={tracks}
+      user={user}
+    />
+  )
+}
+
+const CollectionTileComponent = ({
+  collection,
+  togglePlay,
+  tracks,
+  user,
+  ...props
+}: LineupItemProps & {
+  collection: Collection
+  tracks: EnhancedCollectionTrack[]
+  user: User
+}) => {
+  const dispatchWeb = useDispatchWeb()
+  const navigation = useNavigation()
+  const currentUserId = useSelectorWeb(getUserId)
+
+  const {
+    has_current_user_reposted,
+    has_current_user_saved,
+    is_album,
+    playlist_id,
+    playlist_name,
+    playlist_owner_id
+  } = collection
+
+  const isOwner = playlist_owner_id === currentUserId
+
+  const routeWeb = useMemo(() => {
+    return collection.is_album
+      ? albumPage(user.handle, collection.playlist_name, collection.playlist_id)
+      : playlistPage(
+          user.handle,
+          collection.playlist_name,
+          collection.playlist_id
+        )
+  }, [collection, user])
+
+  const handlePress = useCallback(() => {
+    const trackUid = tracks[0] ? tracks[0].uid : null
+    const trackId = tracks[0] ? tracks[0].track_id : null
+    if (!trackUid || !trackId) {
+      return
+    }
+    togglePlay(trackUid, trackId, PlaybackSource.PLAYLIST_TILE_TRACK)
+  }, [togglePlay, tracks])
+
+  const handlePressTitle = useCallback(() => {
+    navigation.push({
+      // TODO: update to `collection` screen
+      native: { screen: 'track', params: { id: playlist_id } },
+      web: { route: routeWeb }
+    })
+  }, [playlist_id, routeWeb, navigation])
+
+  const duration = useMemo(() => {
+    return tracks.reduce(
+      (duration: number, track: Track) => duration + track.duration,
+      0
+    )
+  }, [tracks])
+
+  const handlePressOverflow = useCallback(() => {
+    if (playlist_id === undefined) {
```

Comment:
so is it reasonable for playlist_id to actually be missing?

Reply:
Not sure actually, but this came from existing code and I didn't want to risk breaking things